src/vehicle.cpp

```diff
@@ -3694,6 +3697,81 @@ bool vehicle::sufficient_wheel_config() const
     return true;
 }
 
+bool vehicle::handle_potential_theft( player *p, bool check_only, bool prompt )
```

Comment:
```suggestion
bool vehicle::handle_potential_theft( player &p, bool check_only, bool prompt )
```

You don't even handle the case when `p` is null, so make it clear to the caller that they must supply a *valid* player reference.

Reply:
or check for nullptr in the function? or is a reference preferable?

Reply:
How would you handle a null pointer there? A debugmsg? By requiring a reference, you avoid the need for this check and let the caller do it (if needed).

Reply:
> You don't even handle the case when `p` is null, so make it clear to the caller that they must supply a _valid_ player reference.

Well, I've gone for nullptr checking, hope thats suitable?

Reply:
> How would you handle a null pointer there? A debugmsg? By requiring a reference, you avoid the need for this check and let the caller do it (if needed).

Hmm, well a null pointer shouldnt happen ( famous last words, I assume ) and if it does, a debugmsg would probably be suitable, and a return false, would just dump player back to what they were doing before they requested to interact with the vehicle.

Reply:
I mean I don't know enough about best practices in C++ to know one way or another, I just know I seen player pointers used everywhere in trapfunc and activity code and so on, so i assumed it was the done thing, But if a reference is better, then I can do that.

Reply:
> well a null pointer shouldnt happen 

Than don't allow it. Use references and there will be no null pointers - there *can* be no null pointer, so you don't have to check, don't have to document, don't have to handle it.

>  I just know I seen player pointers 

Sometimes pointers are useful, doesn't mean you have to use them everywhere.